lib/views/git-revision-view.coffee

```diff
@@ -0,0 +1,124 @@
+_ = require 'underscore-plus'
+path = require 'path'
+fs = require 'fs'
+
+{CompositeDisposable, BufferedProcess} = require "atom"
+{$} = require "atom-space-pen-views"
+
+SplitDiff = require 'split-diff'
+
+
+module.exports =
+class GitRevisionView
+
+  @FILE_PREFIX = "Git Plus - "
+
+  @showRevision: (editor, gitRevision, options={}) ->
+    options = _.defaults options,
+      diff: false
+
+    SplitDiff.disable(false)
+
+    file = editor.getPath()
+
+    fileContents = ""
+    stdout = (output) =>
+      fileContents += output
+    exit = (code) =>
+      if code is 0 || options.type is "D"
+        @_showRevision(file, editor, gitRevision, fileContents, options)
+      else
+        atom.notifications.addError "Could not retrieve revision for #{path.basename(file)} (#{code})"
+
+    showArgs = ["show", "#{gitRevision}:./#{path.basename(file)}"]
+    process = new BufferedProcess({
+      command: "git",
+      args: showArgs,
+      options: { cwd:path.dirname(file) },
+      stdout,
+      exit
+    })
+
+  @_getInitialLineNumber: (editor) ->
+    editorEle = atom.views.getView editor
+    lineNumber = 0
+    if editor? && editor != ''
+      lineNumber = editorEle.getLastVisibleScreenRow()
+      return lineNumber - 5
+
+
+  @_showRevision: (file, editor, gitRevision, fileContents, options={}) ->
+    outputDir = "#{atom.getConfigDirPath()}/git-plus"
+    fs.mkdir outputDir if not fs.existsSync outputDir
+    outputFilePath = "#{outputDir}/#{@FILE_PREFIX}#{path.basename(file)}"
```

Comment:
I think it would be great to include the branch name in the title. Maybe even replace 'Git Plus' with it